codes/utils.py

```
@@ -102,4 +102,29 @@ def extract_code_from_content(content):
+        if role == "system":
+            model_messages.append(ai_messages.ModelRequest(parts=[ai_messages.SystemPromptPart(content)]))
+        elif role == "user":
+            model_messages.append(ai_messages.ModelRequest.user_text_prompt(content))
+        elif role == "assistant":
+            model_messages.append(ai_messages.ModelResponse(parts=[ai_messages.TextPart(content)]))
+    return model_messages
+
+
+def response_to_dict(response):
+    """Serialize a pydantic_ai ModelResponse to a standard dictionary."""
+    import dataclasses
```

Pylint (3.3.7)

[convention] Import outside toplevel (dataclasses)

(C0415)